mods/cnc/maps/shellmap/shellmap.lua

```diff
@@ -1,3 +1,362 @@
+		unit.Destroy()
+	end
+	if unit.HasProperty("AcceptsUpgrade") and unit.AcceptsUpgrade("unkillable") then
+		unit.GrantUpgrade("unkillable")
+		if unit.Stance ~= nil then
+			unit.Stance = "Defend"
+		end
+	end
+end
+--Repeated Functions END
+
+--Event Sequence Functions START
+SendCommandos = function()
+	local commandos = Reinforcements.ReinforceWithTransport(gdi, "tran", CommandoTeam, 
+	{ChinEntry.Location, ChinUnload.Location}, {ChinExit.Location})[2]
+	
+	Trigger.OnIdle(commandos[1], function()
+			if(Map.NamedActor("Turret1") ~= nil) then
+				commandos[1].Demolish(Turret1)
+			end
+			if(Map.NamedActor("beachdude1") ~= nil or Map.NamedActor("beachdude2") ~= nil) then
+				commandos[1].AttackMove(CommMove.Location)
+			end
+			if(Map.NamedActor("Turret3") ~= nil) then
+				commandos[1].Demolish(Turret3)
+				commandos[1].Move(CommFin.Location)
+			end
+			commandos[1].Move(CommFin.Location)
+	end)
+
+	Trigger.OnIdle(commandos[2], function()
+			if(Map.NamedActor("Turret2") ~= nil) then
+				commandos[2].Demolish(Turret2)
+			end
+			if(Map.NamedActor("beachdude1") ~= nil or Map.NamedActor("beachdude2") ~= nil) then
+				commandos[2].AttackMove(CommMove.Location)
+			end
+			if(Map.NamedActor("Turret4") ~= nil) then
+				commandos[2].Demolish(Turret4)
+				commandos[2].Move(CommFin.Location)
+			end
+			commandos[2].Move(CommFin.Location)
+	end)
+
+	Trigger.OnKilled(Turret1, function()
+			beachdude1.AttackMove(BeachDudeMove.Location)
+			beachdude2.AttackMove(BeachDudeMove.Location)
+	end)
+
+	Trigger.OnKilled(Turret3, function()
+			commandos[1].Stop()
+			commandos[1].ScriptedMove(CommFin.Location)
+			Trigger.Clear(commandos[1], "OnIdle")
+			Trigger.AfterDelay(25, Beachhead)
+	end)
+
+	Trigger.OnKilled(Turret4, function()
+			commandos[2].Stop()
+			commandos[2].ScriptedMove(CommFin.Location)
+			Trigger.Clear(commandos[2], "OnIdle")
+	end)
+	Trigger.OnAllKilled({beachdude1, beachdude2}, function()
+		local trans = Reinforcements.ReinforceWithTransport(gdi, "tran", nil, {EvacEntry.Location, EvacLand.Location})[1]
+		Trigger.OnIdle(trans, function()
+			if(trans.HasPassengers) then
+				trans.Wait(25)
+				trans.CallFunc(function() PlayMusic() end)
+				trans.Move(EvacEntry.Location)
+				trans.Destroy()
+			else
+				commandos[1].ScriptedMove(FlareHere.Location)
+				commandos[2].ScriptedMove(FlareHere.Location)
+				commandos[1].EnterTransport(trans)
+				commandos[2].EnterTransport(trans)
+			end
+		end)
+	end)
+end
+
+Beachhead = function()
+	--todo: Make custom reinforce function to handle LST Dropoff properly
+	local ltank = Reinforcements.ReinforceWithTransport(gdi, "lst", {"htnk"}, LSTLEntryPath, LSTLExitPath, UnloadLST)[2]
+	Trigger.OnAddedToWorld(ltank[1], function(unit) LeftTankAction(unit) end)
+	Trigger.AfterDelay(55, function()
+		local mcv = Reinforcements.ReinforceWithTransport(gdi, "lst", {"mcv"}, LSTMEntryPath, LSTMExitPath, UnloadLST)[2]
+		Trigger.OnAddedToWorld(mcv[1], function(unit) MCVAction(unit) end)
```

Comment:
I'll clean up a bit with this information. I came to the conclusion that I needed a new function since I was calling it with parentheses and getting nothing but errors.